internal/helm/controller/controller_test.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+func TestFilterPredicate(t *testing.T) {
+	selector := make(map[string]string)
+	selector["testKey"] = "testValue"
+	options := WatchOptions{
+		Selector: metav1.LabelSelector{
+			MatchLabels: selector,
+		},
+		GVK: schema.GroupVersionKind{
+			Group:   "test-group",
+			Version: "v1",
+			Kind:    "Test-Kind",
+		}}
+	cfg, _ := config.GetConfig()
+	mgr, _ := manager.New(cfg, manager.Options{})
```

Comment:
I'm surprised this returns a valid manager. @VenkatRamaraju does this pass locally for you?

Reply:
Ah it [does not](https://github.com/operator-framework/operator-sdk/pull/4997/checks?check_run_id=3194157925#step:4:288) pass. I recommend refactoring the predicate parse logic into its own function and writing a test specifically for that function.

Reply:
@estroz, yeah it does. `make test-unit` passes locally too. 
```
$ go test -v 
=== RUN   TestFilterPredicate
--- PASS: TestFilterPredicate (0.28s)
=== RUN   TestHasAnnotation
--- PASS: TestHasAnnotation (0.00s)
PASS
ok  	github.com/operator-framework/operator-sdk/internal/helm/controller	1.038s
```